include/deal.II/base/mpi.h

```diff
@@ -1762,6 +1792,40 @@ namespace Utilities
 
 
 
+    template <typename T>
+    void
+    broadcast(T *                buffer,
+              const size_t       count,
+              const unsigned int root,
+              const MPI_Comm &   comm)
+    {
+#  ifndef DEAL_II_WITH_MPI
+      (void)buffer;
+      (void)count;
+      (void)root;
+      (void)comm;
+#  else
+      // MPI_Bcast's count is a signed int, so send at most 2^31 in each
+      // iteration:
+      const size_t max_send_count = std::numeric_limits<signed int>::max();
+
+      size_t total_sent_count = 0;
+      while (total_sent_count < count)
+        {
+          const size_t current_count =
+            std::min(count - total_sent_count, max_send_count);
+
+          const int ierr =
+            MPI_Bcast(buffer, current_count, mpi_type_id(buffer), root, comm);
```

Comment:
I think it would be easier to read if you wrote this as
```
MPI_Bcast(buffer + total_send_count, current_count, mpi_type_id(buffer), root, comm);
```
because then `buffer` really always points to the originally provided `buffer`. This way you can also make the argument `const`, and you don't have to increment it below.

Reply:
No, if you don't ever change the `buffer` pointer, you can make it `const T *` (or `const T * const` if you so want).

Reply:
No, that won't work because we are writing into ``buffer`` on the other ranks.

Reply:
Oh yes, I get it now!